### ISSUE TITLE
use wasInformedBy to relate activities

### ISSUE THREAD

**Issue by jhpoelen** [bio-guoda/preston#34]
Currently, we use "activity" verb to 

```
<285cef08-fcfc-4acd-bf51-c1f73d283f72> <http://www.w3.org/ns/prov#activity> <124b8bdb-00f4-4633-8f16-1d4108a67a63> .
```

Instead, we should use something like:

```
<285cef08-fcfc-4acd-bf51-c1f73d283f72> <http://www.w3.org/ns/prov#wasInformedBy> <124b8bdb-00f4-4633-8f16-1d4108a67a63> .
```

**Comment by jhpoelen** [comment]
Also, should patch the existing logs to rewrite the ```#activity``` relations.

**Comment by mielliott** [comment]
"activity" verb changed to "wasInformedBy" for generation events in d29c26785ebd639f0aa79cb5ef35390bf4c35068. That seems to be the only place we were using "activity". @jhpoelen confirm?

Working on the patch.

**Comment by mielliott** [comment]
See https://github.com/bio-guoda/preston-scripts/blob/master/patch-generation-activities.py.

Since the only place we currently use "activity" is in the generations, the script simply replaces all occurrences of "activity" with "wasInformedBy". The patch can be applied to existing preston observatories like so:
```shell
preston ls | python3 patch-generation-activities.py | preston process
```

**Comment by jhpoelen** [comment]
Just patched the preston GBIF/iDigBio/BioCASe with provenance log version:

hash://sha256/3b39831bcc286c1db44787e21b736378f5847a16b7c39bdac3dd2011e9189dc1

```
$ preston cat hash://sha256/3b39831bcc286c1db44787e21b736378f5847a16b7c39bdac3dd2011e9189dc1 | tail -n+10 | head
<https://doi.org/10.5281/zenodo.1410543> <http://purl.org/dc/terms/bibliographicCitation> "Jorrit Poelen, Icaro Alzuru, & Michael Elliott. 2019. Preston: a biodiversity dataset tracker (Version 0.1.10) [Software]. Zenodo. http://doi.org/10.5281/zenodo.1410543"@en .
<0659a54f-b713-4f86-a917-5be166a14110> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://www.w3.org/ns/prov#Entity> .
<0659a54f-b713-4f86-a917-5be166a14110> <http://purl.org/dc/terms/description> "A biodiversity dataset graph archive."@en .
<hash://sha256/916255b2b73680595dcb22b30991a757dd223208473fb4fbe90405757bc07953> <http://www.w3.org/ns/prov#usedBy> <8182923e-060d-41e4-b70d-7a3c963e766e> .
<5461c6a9-7780-45c7-9b22-74777abcd364> <http://www.w3.org/ns/prov#wasInformedBy> <501dadd3-0468-4a12-b6a9-1cd6c228be70> .
<cda0b051-8f57-4462-b8e5-68d7417a5074> <http://www.w3.org/ns/prov#wasInformedBy> <501dadd3-0468-4a12-b6a9-1cd6c228be70> .
```

Other patching in progress . . .

**Comment by jhpoelen** [comment]
Preston ALA observatory was patched with:

```
$ preston cat hash://sha256/96b314d9a5d39b8a722a75a0cba943b682804f0bb34024ef1e3565abe7495550 | head
<https://preston.guoda.bio> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://www.w3.org/ns/prov#SoftwareAgent> .
<https://preston.guoda.bio> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://www.w3.org/ns/prov#Agent> .
<https://preston.guoda.bio> <http://purl.org/dc/terms/description> "Preston is a software program that finds, archives and provides access to biodiversity datasets."@en .
<872d7519-3e30-4cdd-8d4d-5757297a10b0> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://www.w3.org/ns/prov#Activity> .
<872d7519-3e30-4cdd-8d4d-5757297a10b0> <http://purl.org/dc/terms/description> "An event that (re-) processes existing biodiversity datasets graphs and their provenance."@en .
<872d7519-3e30-4cdd-8d4d-5757297a10b0> <http://www.w3.org/ns/prov#startedAtTime> "2020-03-09T18:42:15.365Z"^^<http://www.w3.org/2001/XMLSchema#dateTime> .
<872d7519-3e30-4cdd-8d4d-5757297a10b0> <http://www.w3.org/ns/prov#wasStartedBy> <https://preston.guoda.bio> .
<https://doi.org/10.5281/zenodo.1410543> <http://www.w3.org/ns/prov#usedBy> <872d7519-3e30-4cdd-8d4d-5757297a10b0> .
<https://doi.org/10.5281/zenodo.1410543> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://purl.org/dc/dcmitype/Software> .
<https://doi.org/10.5281/zenodo.1410543> <http://purl.org/dc/terms/bibliographicCitation> "Jorrit Poelen, Icaro Alzuru, & Michael Elliott. 2019. Preston: a biodiversity dataset tracker (Version 0.1.10) [Software]. Zenodo. http://doi.org/10.5281/zenodo.1410543"@en .
```

**Comment by jhpoelen** [comment]
Preston OBIS observatory - 

```
$ preston cat hash://sha256/3970258f96cd03d3d384ebdd664f1c7fd8f293913e2e1ef3bc130fe3fed75ca2 | head
<https://preston.guoda.bio> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://www.w3.org/ns/prov#SoftwareAgent> .
<https://preston.guoda.bio> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://www.w3.org/ns/prov#Agent> .
<https://preston.guoda.bio> <http://purl.org/dc/terms/description> "Preston is a software program that finds, archives and provides access to biodiversity datasets."@en .
<f98d2f87-98b9-44b6-a728-5b223eb11b1b> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://www.w3.org/ns/prov#Activity> .
<f98d2f87-98b9-44b6-a728-5b223eb11b1b> <http://purl.org/dc/terms/description> "An event that (re-) processes existing biodiversity datasets graphs and their provenance."@en .
<f98d2f87-98b9-44b6-a728-5b223eb11b1b> <http://www.w3.org/ns/prov#startedAtTime> "2020-03-09T18:45:14.167Z"^^<http://www.w3.org/2001/XMLSchema#dateTime> .
<f98d2f87-98b9-44b6-a728-5b223eb11b1b> <http://www.w3.org/ns/prov#wasStartedBy> <https://preston.guoda.bio> .
<https://doi.org/10.5281/zenodo.1410543> <http://www.w3.org/ns/prov#usedBy> <f98d2f87-98b9-44b6-a728-5b223eb11b1b> .
<https://doi.org/10.5281/zenodo.1410543> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://purl.org/dc/dcmitype/Software> .
<https://doi.org/10.5281/zenodo.1410543> <http://purl.org/dc/terms/bibliographicCitation> "Jorrit Poelen, Icaro Alzuru, & Michael Elliott. 2019. Preston: a biodiversity dataset tracker (Version 0.1.10) [Software]. Zenodo. http://doi.org/10.5281/zenodo.1410543"@en .
```

**Comment by jhpoelen** [comment]
Preston DataONE observatory (after fix and upgrade to Preston v0.1.11):

```
$ preston cat hash://sha256/2910f784f84e112f124a56ce54bd06b76e510f90276629d2d144ce29e326d80f | head -n20
<https://preston.guoda.bio> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://www.w3.org/ns/prov#SoftwareAgent> .
<https://preston.guoda.bio> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://www.w3.org/ns/prov#Agent> .
<https://preston.guoda.bio> <http://purl.org/dc/terms/description> "Preston is a software program that finds, archives and provides access to biodiversity datasets."@en .
<ff67bbc8-fae0-437c-8133-ed46229be9a7> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://www.w3.org/ns/prov#Activity> .
<ff67bbc8-fae0-437c-8133-ed46229be9a7> <http://purl.org/dc/terms/description> "An event that (re-) processes existing biodiversity datasets graphs and their provenance."@en .
<ff67bbc8-fae0-437c-8133-ed46229be9a7> <http://www.w3.org/ns/prov#startedAtTime> "2020-03-09T20:32:13.532Z"^^<http://www.w3.org/2001/XMLSchema#dateTime> .
<ff67bbc8-fae0-437c-8133-ed46229be9a7> <http://www.w3.org/ns/prov#wasStartedBy> <https://preston.guoda.bio> .
<https://doi.org/10.5281/zenodo.1410543> <http://www.w3.org/ns/prov#usedBy> <ff67bbc8-fae0-437c-8133-ed46229be9a7> .
<https://doi.org/10.5281/zenodo.1410543> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://purl.org/dc/dcmitype/Software> .
<https://doi.org/10.5281/zenodo.1410543> <http://purl.org/dc/terms/bibliographicCitation> "Jorrit Poelen, Icaro Alzuru, & Michael Elliott. 2019. Preston: a biodiversity dataset tracker (Version 0.1.11) [Software]. Zenodo. http://doi.org/10.5281/zenodo.1410543"@en .
<0659a54f-b713-4f86-a917-5be166a14110> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://www.w3.org/ns/prov#Entity> .
<0659a54f-b713-4f86-a917-5be166a14110> <http://purl.org/dc/terms/description> "A biodiversity dataset graph archive."@en .
<hash://sha256/e54b360a4ca84a4503e4c10a8a8cca062c130be7429c8fe6ea1e0e82fe113e12> <http://www.w3.org/ns/prov#usedBy> <ff67bbc8-fae0-437c-8133-ed46229be9a7> .
<b563ba40-e911-403b-bafd-c0c5fef5a2cc> <http://www.w3.org/ns/prov#wasInformedBy> <97892995-268b-4ab7-8d05-e015511075eb> .
<66fd5f46-56cd-4b86-84a0-053b8faf1371> <http://www.w3.org/ns/prov#wasInformedBy> <97892995-268b-4ab7-8d05-e015511075eb> .
```

**Comment by jhpoelen** [comment]
Preston BHL observatory (after upgrade to preston v0.1.11) -

```
$ preston cat hash://sha256/9215d543418a80510e78d35a0cfd7939cc59f0143d81893ac455034b5e96150a | head -n15
<https://preston.guoda.bio> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://www.w3.org/ns/prov#SoftwareAgent> .
<https://preston.guoda.bio> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://www.w3.org/ns/prov#Agent> .
<https://preston.guoda.bio> <http://purl.org/dc/terms/description> "Preston is a software program that finds, archives and provides access to biodiversity datasets."@en .
<eb5b8cf5-1969-4773-8c6e-a4aff354aaab> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://www.w3.org/ns/prov#Activity> .
<eb5b8cf5-1969-4773-8c6e-a4aff354aaab> <http://purl.org/dc/terms/description> "An event that (re-) processes existing biodiversity datasets graphs and their provenance."@en .
<eb5b8cf5-1969-4773-8c6e-a4aff354aaab> <http://www.w3.org/ns/prov#startedAtTime> "2020-03-09T20:40:10.062Z"^^<http://www.w3.org/2001/XMLSchema#dateTime> .
<eb5b8cf5-1969-4773-8c6e-a4aff354aaab> <http://www.w3.org/ns/prov#wasStartedBy> <https://preston.guoda.bio> .
<https://doi.org/10.5281/zenodo.1410543> <http://www.w3.org/ns/prov#usedBy> <eb5b8cf5-1969-4773-8c6e-a4aff354aaab> .
<https://doi.org/10.5281/zenodo.1410543> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://purl.org/dc/dcmitype/Software> .
<https://doi.org/10.5281/zenodo.1410543> <http://purl.org/dc/terms/bibliographicCitation> "Jorrit Poelen, Icaro Alzuru, & Michael Elliott. 2019. Preston: a biodiversity dataset tracker (Version 0.1.11) [Software]. Zenodo. http://doi.org/10.5281/zenodo.1410543"@en .
<0659a54f-b713-4f86-a917-5be166a14110> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://www.w3.org/ns/prov#Entity> .
<0659a54f-b713-4f86-a917-5be166a14110> <http://purl.org/dc/terms/description> "A biodiversity dataset graph archive."@en .
<hash://sha256/e46cd4b0d7fdb51ea789fa3c5f7b73591aca62d2d8f913346d71aa6cf0745c9f> <http://www.w3.org/ns/prov#usedBy> <eb5b8cf5-1969-4773-8c6e-a4aff354aaab> .
<6c69926d-179d-41d7-a0df-cd7e47cf7ad9> <http://www.w3.org/ns/prov#wasInformedBy> <d8afa467-0bd4-4fa8-b618-2944e96bb7d0> .
<7b1746e5-cc28-44c6-933f-453f49a6b76f> <http://www.w3.org/ns/prov#wasInformedBy> <d8afa467-0bd4-4fa8-b618-2944e96bb7d0> .
```

**Comment by jhpoelen** [comment]
@mielliott all observatories have been patched . . . as I was patching I found, reported and fixed issue #40 . This bug affected DataONE and BHL observatories in period Dec 2019 - Mar 2020 , meaning that while content was being observed, the provenance logs in that period were not saved.

**Comment by jhpoelen** [comment]
This completes the fixing and patching related to this issue (activity -> wasInformedBy). Please re-open issue if more work is needed or questions remain.
